lib/makeVariantStyles.ts

```diff
@@ -0,0 +1,51 @@
+import { cva } from 'class-variance-authority';
+import {CompoundVariantsType, InteractionProps} from "@/types/colors";
+
+export const defaultVariantStyles = {
+  variants: {
+    intent: {
+      primary: [''],
+      secondary: [''],
+      neutral: [''],
+      success: [''],
+      danger: [''],
+      info: ['']
+    },
+    mode: {
+      default: [''],
+      dark: ['']
+    },
+    interaction: {
+      hover: [''],
+      active: [''],
+      disabled: ['']
+    },
+    prominence: {
+      default: [''],
+      spicy: [''],
+      mild: ['']
+    }
+  }
+};
+
+export const getInteraction = ({ active, disabled }: InteractionProps) =>
```

Comment:
probably worth refactoring this chained ternary just for readability

seems like:
1. if the action is active, return 'active'
2. else, if the action is disabled, return 'disabled'
3. else return 'hover'

without looking at how it's implemented in practice, It's hard to grok what this helper does (and why the default would be 'hover' without comments.